tests/test_caching.py

```diff
@@ -684,14 +684,16 @@ def test_sparse_function(self, operate_on_empty_cache):
         # in the first clear_cache they were still referenced by their "parent" objects
         # (e.g., ru* by ConditionalDimensions, through `condition`)
 
-        assert len(_SymbolCache) == init_cache_size + 8
+        assert len(_SymbolCache) == init_cache_size + 12
         clear_cache()
         # Now we should be back to the original state except for
-        # pos* that belong to the abstract class
-        assert len(_SymbolCache) == init_cache_size + 2
+        # pos* that belong to the abstract class and the dimension
+        # bounds (x_m, x_M, y_m, y_M)
+        assert len(_SymbolCache) == init_cache_size + 6
         clear_cache()
-        # Now we should be back to the original state
-        assert len(_SymbolCache) == init_cache_size
+        # Now we should be back to the original state plus the dimension bounds
+        # (x_m, x_M, y_m, y_M)
+        assert len(_SymbolCache) == init_cache_size + 4
```

Comment:
That's not good, why is there a leftover 4?

Reply:
`x_m, x_M, y_m, y_M` are not longer touched at grid instantiation, and so are only created and cached at `i = u.inject(expr=u, field=u)`. However, the grid and subdimension thickness symbols persist and thus these symbols persist in the cache to my understanding

Reply:
But is there a  cache leak? If it needs one more cache clear to remove it that's ok (and you should add an extra line testing it gets back to `init_cache_size ` after one more) but if there is a reference that cannot be reached then it's problematic.

Reply:
How is it determined that an object should be removed from cache? If there's no other objects referencing it?

Reply:
Yes. The cache is no different than the standard python GC, it clears objects when there are no more references to it, so if it cannot be cleared then there is unreachable references

Reply:
Ok, added additional testing to check that all symbols disappear once the grid goes out of scope